Sources/SwiftJWT/BlueECDSA.swift

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright IBM Corporation 2018
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+import CryptorECC
+import LoggerAPI
+
+import Foundation
+
+class BlueECSigner: SignerAlgorithm {
+    let name: String = "ECDSA"
+    
+    private let key: Data
+    private let curve: String
+    
+    init(key: Data, curve: String) {
+        self.key = key
+        self.curve = curve
+    }
+    
+    func sign(header: String, claims: String) throws -> String {
+        let unsignedJWT = header + "." + claims
+        guard let unsignedData = unsignedJWT.data(using: .utf8) else {
+            throw JWTError.invalidJWTString
+        }
+        let signature = try sign(unsignedData)
+        let signatureString = signature.base64urlEncodedString()
+        return header + "." + claims + "." + signatureString
+    }
+    
+    func sign(_ data: Data) throws -> Data {
+        guard let keyString = String(data: key, encoding: .utf8) else {
+            throw JWTError.invalidPrivateKey
+        }
+        if #available(OSX 10.13, *) {
+            let privateKey = try ECPrivateKey(key: keyString)
+            guard privateKey.curveId == curve else {
```

Comment:
Damn, ECPrivateKey.curveId should have been an enum (or struct equivalent) - is it too late to change it?